waiter/src/waiter/request_log.clj

```diff
@@ -93,7 +93,11 @@
 (defn log-request!
   "Log a request"
   [request response]
-  (log (merge (request->context request) (response->context response))))
+  (let [redacted-request-fields-string (get-in response [:descriptor :service-description "env" "REDACTED_REQUEST_FIELDS"])]
+    (log (apply dissoc
+                (merge (request->context request) (response->context response))
+                (when-not (str/blank? redacted-request-fields-string)
```

Comment:
Rather than `(apply dissoc ... (when-not ...))`, I think `(cond->> ... (not ...) (apply dissoc ...))` is more idiomatic.